v20.1/cockroachcloud-upgrade-to-v20.1.md

```diff
@@ -0,0 +1,161 @@
+
+When you start the upgrade, the cluster will be briefly unavailable while the node is stopped and restarted with v20.1.
+
+Approximately 24 hours after the node has been restarted, the upgrade will be automatically finalized. This enables certain [features and performance improvements introduced in v20.1](#respect-temporary-limitations). Finalization also removes the ability to roll back to v19.2, so it's important to monitor your application during this 24-hour window and, if you see unexpected behavior, trigger a rollback from the CockroachCloud Console. Also, there are some [temporary limitations](#review-temporary-limitations) during this 24-hour window, so if everything looks good, you'll have the choice to finalize the upgrade more quickly so as to lift these limitations.
+
+</section>
+
+## Step 3. Prepare to upgrade
+
+ Before starting the upgrade, it's important to complete the following steps.
+
+<section class="filter-content" markdown="1" data-scope="single-node">
+
+### Prepare for brief unavailability
+
+Because your cluster will be unavailable while its single node is stopped and restarted with v20.1, prepare your application for this brief downtime.
+
+</section>
+
+### Review breaking changes
+
+Review the following list of backward-incompatible changes in v20.1, and if any affect your application, make necessary changes:
+
+- The `extract()` [built-in function](functions-and-operators.html) with sub-second arguments (millisecond, microsecond) is now Postgres-compatible and returns the total number of seconds in addition to sub-seconds instead of returning only sub-seconds.
+
+- Casting intervals to integers and floats is now Postgres-compatible and values a year at 365.25 days in seconds instead of 365 days.
+
+- The combination of the [`CHANGEFEED`](change-data-capture.html) options `format=experimental_avro`, `envelope=key_only`, and `updated` is now rejected. This is because the use of `key_only` prevents any rows with updated fields from being emitted, which renders the `updated` option meaningless.
+
+- The `cockroach user` CLI command has been removed. It was previously deprecated in CockroachDB v19.2. Note that a v19.2 client (supporting `cockroach user`) can still operate user accounts in a v20.1 server.
+
+- The [`GRANT`](grant.html) and [`REVOKE`](revoke.html) statements now require that the requesting user already have the target privileges themselves. For example, `GRANT SELECT ON t TO foo` requires that the requesting user already have the `SELECT` privilege on `t`.
+
+### Let ongoing bulk operations finish
+
+Make sure there are no [bulk imports](import.html) or [schema changes](online-schema-changes.html) in progress. These are complex operations that can increase the potential for unexpected behavior during an upgrade.</span>
+
+To check for ongoing bulk operations, use [`SHOW JOBS`](show-jobs.html#show-schema-changes) or check the [**Jobs** page](admin-ui-jobs-page.html) in the Admin UI.
```

Comment:
The link to `show-jobs.html` here doesn't put us in the right section (show-schema-changes). Do we know why?

Reply:
Hmm, not sure why that happens. The anchor is actually stripped from the rendered page somehow. For now, I'll remove the anchor portion.